konig-dao-core/src/main/java/io/konig/dao/core/FieldPath.java

```diff
@@ -87,31 +87,42 @@ public static FieldPath createFieldPath(String path, EntityStructure struct) thr
 	/**
 	 * Build a path to the first field in the given structure that has the MEASURE stereotype.
 	 */
-	public static FieldPath measurePath(EntityStructure struct) {
+	public static FieldPath measurePath(EntityStructure struct, String yAxis) {
 		FieldPath result = new FieldPath(null);
-		return measurePath(result, struct) ? result : null;
+		return measurePath(result, struct, yAxis) ? result : null;
 	}
 	
-	public static FieldPath dimensionPath(EntityStructure struct) {
+	public static FieldPath dimensionPath(EntityStructure struct, String xAxis) {
 		FieldPath result = new FieldPath(null);
-		return dimensionPath(result, struct) ? result : null;
+		return dimensionPath(result, struct, xAxis) ? result : null;
 	}
 	
-	private static boolean dimensionPath(FieldPath result, EntityStructure struct) {		
+	private static boolean dimensionPath(FieldPath result, EntityStructure struct, String xAxis) {		
 		for (FieldInfo field : struct.getFields()) {	
 			if (Stereotype.DIMENSION.equals(field.getStereotype()) 
-					&& field.getFieldType().getNamespace().equals("http://schema.org/")) {
+					&& xAxis.endsWith(field.getName())) {
 				result.add(field);
 				return true;
 			}
 		}
+		for (FieldInfo field : struct.getFields()) {
+			EntityStructure child = field.getStruct();
+			if (child != null) {
+				result.add(field);
+				if (dimensionPath(result, child, xAxis)) {
+					return true;
+				}
+				result.pop();
+			}
+		}
 		return false;
 	}
 	
-	private static boolean measurePath(FieldPath result, EntityStructure struct) {
+	private static boolean measurePath(FieldPath result, EntityStructure struct, String yAxis) {

```

Comment:
This implementation is risky.  
Instead of testing yAxis.endsWith(field.getName()), you should construct the full, dot-separated field name as you descend the EntityStructure and then check for an exact match.

If you don't know how to do this, let me know and I'll send you a code snippet.